Libraries/LibChess/Chess.h

```diff
@@ -105,6 +105,14 @@ struct Move {
     Square from;
     Square to;
     Type promote_to;
+    Piece piece;
+    bool is_check = false;
+    bool is_mate = false;
+    bool is_capture = false;
+    bool is_castle_short = false;
+    bool is_castle_long = false;
```

Comment:
Castling can be determined by `from` `to` and `piece`. I think it would be better to make these methods based on those.

Reply:
I didn't want to duplicate that castling logic, so I ended up setting those flags where castling would be calculated anyway.
It would be nice to debloat the `Move` struct by removing some flags though.
So I'm thinking of separating the castling check to it's own method and then call it from `apply_illegal_move()` and `Move::to_algebraic()`, removing the need for the bools.

Reply:
Okay, since the logic in the other method is not quite the same, and the logic we need here is quite simple I just added the following at the start of `Move::to_algebraic()`:
```c++
if (piece.type == Type::King && from.file == 4) {
    if (to.file == 2)
        return "O-O-O";
    if (to.file == 6)
        return "O-O";
}
```